lib/ansible/module_utils/facts/system/service_mgr.py

```diff
@@ -52,6 +52,16 @@ def is_systemd_managed(module):
             for canary in ["/run/systemd/system/", "/dev/.run/systemd/", "/dev/.systemd/"]:
                 if os.path.exists(canary):
                     return True
+
+            # If all else fails, check if init is the systemd command, using comm as cmdline could be symlink
+            try:
+                with open('/proc/1/comm', 'r') as init_proc:
+                    init = init_proc.readline().strip()
+                    return init == 'systemd'
+            except IOError:
+                # If comm doesn't exist, old kernel, no systemd
+                return False
+
```

Comment:
i would cenetralize in module_utils/service.py instead

Reply:
Ok.